silo/views.py

```diff
@@ -936,9 +936,13 @@ def getJSON(request):
 def toggle_silo_publicity(request):
     silo_id = request.GET.get('silo_id', None)
     silo = Silo.objects.get(pk=silo_id)
-    silo.public = not silo.public
-    silo.save()
-    return HttpResponse("Your change has been saved")
+
+    if silo.owner == request.user:
+        silo.public = not silo.public
+        silo.save()
+        return HttpResponse("Your change has been saved.")
+    else:
+        return HttpResponse("You have not permission to change publicity.")
```

Comment:
- Why did you decide to return the default status code?
- The message isn't shown to the users. Shouldn't we give a feedback?